lib/github_advisory_sync.rb

```diff
@@ -199,110 +208,120 @@ def ghsa_id
-          "cvss_v3" => cvss,
-          "patched_versions" => [ "<FILL IN SEE BELOW>" ],
-          "unaffected_versions" => [ "<OPTIONAL: FILL IN SEE BELOW>" ]
-        }
-        data["cve"] = cve_id[4..20] if cve_id
-
-        dir_to_write = File.dirname(filename_to_write)
-        Dir.mkdir dir_to_write unless Dir.exist?(dir_to_write)
-        File.open(filename_to_write, "w") do |file|
-          # create an automatically generated advisory yaml file
-          file.write data.to_yaml
-
-          # The data we just wrote is incomplete,
-          # and therefore should not be committed as is
-          # We can not directly translate from GitHub to rubysec advisory format
-          #
-          # The patched_versions field is not exactly available.
-          # - GitHub has a first_patched_version field,
-          #   but rubysec advisory needs a ruby version spec
-          #
-          # The unnaffected_versions field is similarly not directly available
-          # This optional field must be inferred from the vulnerableVersionRange
-          #
-          # To help write those fields, we put all the github data below.
-          #
-          # The second block of yaml in a .yaml file is ignored (after the second "---" line)
-          # This effectively makes this data a large comment
-          # Still it should be removed before the data goes into rubysec
-          file.write "\n\n# GitHub advisory data below - **Remove this data before committing**\n"
-          file.write "# Use this data to write patched_versions (and potentially unaffected_versions) above\n"
-          file.write github_advisory_graphql_object.to_yaml
+      return if saved_data.key?("cvss_v3") || cvss.nil?
+
+      cvss_added = false
+      File.open("#{rubysec_filename}.tmp", "w") do |f| 
+        IO.foreach(rubysec_filename) do |line|
+          if (line.include?('unaffected_versions:') || line.include?('patched_versions:')) && !cvss_added
+            f.write("cvss_v3: #{cvss}\n\n")
+              cvss_added = true
+          end
+          f.write(line)
         end
-        puts "Wrote: #{filename_to_write}"
-        files_written << filename_to_write
       end
 
-      files_written
+      File.delete(rubysec_filename)
+      File.rename("#{rubysec_filename}.tmp", rubysec_filename)
+
+      puts "Updated: #{rubysec_filename}"
+
+      rubysec_filename
+    end
+
+    def write_file
+      return update_file if File.exist?(rubysec_filename)
+
+      data = {
+        "gem" => package_name,
+        "ghsa" => ghsa_id[5..],
+        "url" => external_reference,
+        "date" => published_day,
+        "title" => advisory["summary"],
+        "description" => advisory["description"],
+        "cvss_v3" => cvss,
+        "patched_versions" => ["<FILL IN SEE BELOW>"],
+        "unaffected_versions" => ["<OPTIONAL: FILL IN SEE BELOW>"]
+      }
+      data["cve"] = cve_id[4..20] if cve_id
+      data["cvss_v3"] = "<FILL IN IF AVAILABLE>" if cvss.nil?
```

Comment:
Is this where we should at some future point use NVD's API to check if they have an available score? *cough* https://github.com/rubysec/ruby-advisory-db/issues/399